### PR TITLE
:sparkles: Db integration

### PR DESCRIPTION
**Pull request by amk837**
This PR integrates the `mysql` database using `sequelize` to handle db operation. Redid the whole courses task using db

### CODE REVIEW

File: src/controller/courses/db/index.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+};
+
+const updateCourse = async(req, res) => {
+	try {
+		const [updated] = await coursesModel.update({...req.body}, {
+			where: {id: req.params.id}
+		});
+
+		if (!updated) {
+			res.status(404).send({error: 'No course found with given id'});
```
Comment (ZainAmjad68) [inline src/controller/courses/db/index.js:46]: an alternate flow could be:
- first check if the course exists (using `findByPk`)
- if not, throw an error saying `No Course with given ID`
- if yes, try to update it using the `req.body`
    - if success, return the new updated course
    - if failure, a different error saying the body was invalid
    - OR alternatively, we could try to validate the request body even before trying to update

Comment (amk837) [reply src/controller/courses/db/index.js:43]: Thanks for this comment. I missed this. I have updated the code to reflect the suggested comment

File: src/middlewares/cross-origin/index.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const crossOrigin = (req, res, next) => {
+	res.header('Access-Control-Allow-Origin', '*');
```
Comment (neebz) [inline src/middlewares/cross-origin/index.js:2]: should never be using this

File: src/utils/constants/index.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const {routes} = require('./routes');
```
Comment (neebz) [inline src/utils/constants/index.js:1]: all of folders `utils`, `common`, `shared`, `helpers` - anti-patterns.

File: src/middlewares/cross-origin/index.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+	res.header('Access-Control-Allow-Methods', 'GET, PUT, POST, PATCH, DELETE, SEARCH');
+	res.header('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept, Authorization');
+	next();
+};
+
+module.exports = crossOrigin;
```
Comment (neebz) [inline src/middlewares/cross-origin/index.js:8]: if code is shared between multiple controllers :
- put into middleware (controller).  - routes + middleware + controller
- Or put into model layer

File: src/controller/courses/db/index.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	try {
+		const course = await coursesModel.findByPk(req.params.id);
+		if (!course) {
+			res.status(404).send({error: 'No course found with given id'});
+			return;
+		}
+
+		await course.update({...req.body});
+
+		res.status(200).send(course);
```
Comment (neebz) [inline src/controller/courses/db/index.js:49]: replace `.send` with `.json`

**Review by neebz — APPROVED** [review]

**Pull request merged**